File: java/warmup-1/frontBack.java
// Given a string, return a new string where the first and last chars have been exchanged.

// frontBack("code") → "eodc"
// frontBack("a") → "a"
// frontBack("ab") → "ba"

public String frontBack(String str) {
  String[] newStr = str.split("");
  String first = newStr[0];
  String last = newStr[newStr.length - 1];
  newStr[0] = last;
  newStr[newStr.length - 1] = first;

  return String.join("", newStr);
}

File: python/warmup-1/front_back.py
# Given a string, return a new string where the first and last chars have been exchanged.

# front_back('code') → 'eodc'
# front_back('a') → 'a'
# front_back('ab') → 'ba'

def front_back(str):
    front = str[0:1]
    back = str[len(str)-1:len(str)]
    middle = str[1:len(str)-1]
    if (len(str) < 2):
        return str
    return back + middle + front
